### PR TITLE
Use the correct instrument type while creating aysnchronous instruments

### PR DESCRIPTION
**Pull request by hunguyenaxon**
The fix is for https://github.com/open-telemetry/opentelemetry-swift/issues/471

While creating asynchronous instrument from synchronous one, we need to pass the correct `InstrumentType` instead of using the value of synchronous instance.

### CODE REVIEW

File: Sources/OpenTelemetrySdk/Metrics/Stable/DoubleCounterMeterBuilderSdk.swift [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
     
     public func build() -> OpenTelemetryApi.DoubleCounter {
         buildSynchronousInstrument(DoubleCounterSdk.init)
     }
     
-     public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableDoubleMeasurement) -> Void) -> OpenTelemetryApi.ObservableDoubleCounter {
-        registerDoubleAsynchronousInstrument(type: .counter, updater: callback)
+    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableDoubleMeasurement) -> Void)
+        -> OpenTelemetryApi.ObservableDoubleCounter {
+        registerDoubleAsynchronousInstrument(type: .observableCounter, updater: callback)
```
Comment (hunguyenaxon) [inline Sources/OpenTelemetrySdk/Metrics/Stable/DoubleCounterMeterBuilderSdk.swift:42]: Main change: `.counter` -> `.observableCounter`

File: Sources/OpenTelemetrySdk/Metrics/Stable/DoubleUpDownCounterBuilderSdk.swift [hunk 2]
```diff
@@ -15,28 +15,30 @@ public class DoubleUpDownCounterBuilderSdk : DoubleUpDownCounterBuilder, Instrum
...
     
     public func build() -> OpenTelemetryApi.DoubleUpDownCounter {
         buildSynchronousInstrument(DoubleUpDownCounterSdk.init)
     }
     
-    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableDoubleMeasurement) -> Void) -> OpenTelemetryApi.ObservableDoubleUpDownCounter {
-        registerDoubleAsynchronousInstrument(type: type, updater: callback)
+    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableDoubleMeasurement) -> Void)
+    -> OpenTelemetryApi.ObservableDoubleUpDownCounter {
+        registerDoubleAsynchronousInstrument(type: .observableUpDownCounter, updater: callback)
```
Comment (hunguyenaxon) [inline Sources/OpenTelemetrySdk/Metrics/Stable/DoubleUpDownCounterBuilderSdk.swift:42]: Main change: `type ` -> `. observableUpDownCounter `

File: Sources/OpenTelemetrySdk/Metrics/Stable/LongCounterMeterBuilderSdk.swift [hunk 3]
```diff
@@ -39,10 +37,8 @@ public class LongCounterMeterBuilderSdk : LongCounterBuilder, InstrumentBuilder
         return buildSynchronousInstrument(LongCounterSdk.init)
     }
     
-    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableLongMeasurement) -> Void) -> OpenTelemetryApi.ObservableLongCounter {
-        registerLongAsynchronousInstrument(type: type, updater: callback)
+    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableLongMeasurement) -> Void)
+        -> OpenTelemetryApi.ObservableLongCounter {
+        registerLongAsynchronousInstrument(type: .observableCounter, updater: callback)
```
Comment (hunguyenaxon) [inline Sources/OpenTelemetrySdk/Metrics/Stable/LongCounterMeterBuilderSdk.swift:42]: Main change: `type ` -> `. observableCounter `

File: Sources/OpenTelemetrySdk/Metrics/Stable/LongUpDownCounterBuilderSdk.swift [hunk 4]
```diff
@@ -36,9 +37,8 @@ public class LongUpDownCounterBuilderSdk : LongUpDownCounterBuilder, InstrumentB
         buildSynchronousInstrument(LongUpDownCounterSdk.init)
     }
     
-    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableLongMeasurement) -> Void) -> OpenTelemetryApi.ObservableLongUpDownCounter {
-        registerLongAsynchronousInstrument(type: type, updater: callback)
+    public func buildWithCallback(_ callback: @escaping (OpenTelemetryApi.ObservableLongMeasurement) -> Void)
+        -> OpenTelemetryApi.ObservableLongUpDownCounter {
+        registerLongAsynchronousInstrument(type: .observableUpDownCounter, updater: callback)
```
Comment (hunguyenaxon) [inline Sources/OpenTelemetrySdk/Metrics/Stable/LongUpDownCounterBuilderSdk.swift:42]: Main change: `type ` -> `. observableUpDownCounter `

**Review by vvydier — APPROVED** [review]

**Pull request merged**